File: sandbox.client/front/ts/model/ClientDetail.ts
import {Gender} from "./Gender";
import {ClientAddress} from "./ClientAddress";
import {ClientPhone} from "./ClientPhone";
import {Charm} from "./Charm";

export class ClientDetail {
  public id: number;
  public name: string;
  public surname: string;
  public patronymic: string;
  public gender: Gender;
  public birth_day: Date;
  public charm: Charm;
  public addressFact: ClientAddress;
  public addressReg: ClientAddress;
  public homePhone: ClientPhone;
  public workPhone: ClientPhone;
  public mobilePhone: ClientPhone;

  public assign(o: any): ClientDetail {
    this.id = o.id;
    this.name = o.name;
    this.surname = o.surname;
    this.patronymic = o.patronymic;
    this.gender = o.gender;
    this.birth_day = o.birth_day;
    this.charm = o.charm;
    this.addressFact = o.addressFact;
    this.addressReg = o.addressReg;
    this.homePhone = o.homePhone;
    this.workPhone = o.workPhone;
    this.mobilePhone = o.mobilePhone;
    return this;
  }

  public static copy(a: any): ClientDetail {
    let ret = new ClientDetail();
    ret.assign(a);
    return ret;
  }
}

File: sandbox.controller/src/kz/greetgo/sandbox/controller/model/ClientDetail.java
package kz.greetgo.sandbox.controller.model;

import java.util.Date;

// FIXME: 6/13/18 Details
public class ClientDetail {
  public Integer id;
  public String surname;
  public String name;
  public String patronymic;
  public Gender gender;
  public Date birth_day;
  public Charm charm;
  public ClientAddress addressFact;
  public ClientAddress addressReg;
  public ClientPhone homePhone;
  public ClientPhone workPhone;
  public ClientPhone mobilePhone;

  public ClientDetail() {}

  public ClientDetail(Integer id, String surname, String name, String patronymic, Gender gender, Date birth_day, Charm charm, ClientAddress addressFact, ClientAddress addressReg, ClientPhone homePhone, ClientPhone workPhone, ClientPhone mobilePhone) {
    this.id = id;
    this.surname = surname;
    this.name = name;
    this.patronymic = patronymic;
    this.gender = gender;
    this.birth_day = birth_day;
    this.charm = charm;
    this.addressFact = addressFact;
    this.addressReg = addressReg;
    this.homePhone = homePhone;
    this.workPhone = workPhone;
    this.mobilePhone = mobilePhone;
  }
}
